### PR TITLE
NIAD-215: operation id in inbound gp queue

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package uk.nhs.digital.nhsconnect.nhais.utils;
+
+import com.google.common.hash.Hashing;
+import lombok.NonNull;
+
+import java.nio.charset.StandardCharsets;
+
+public class OperationIdUtils {
```
Comment (grzegorz-robak) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java]: Can we go Object-Oriented way and create just `OperationId` class?
````java
@Getter
public class OperationId {

	private final String value;

	public OperationId(@NonNull String organization, @NonNull Long transactionNumber) {
        this.value = Hashing.sha256()
            .hashString(organization + transactionNumber, StandardCharsets.UTF_8)
            .toString();
	}

	private OperationId(String value) {
		this.value = value;
	}

	public static OperationId random() {
		return OperationId(UUID.randomUUID().toString())
	}
}
````

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package uk.nhs.digital.nhsconnect.nhais.utils;
+
+public interface Headers {
```
Comment (grzegorz-robak) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java]: I'd add context to this name (like `JmsHeaders`)- we might have many different headers :)

Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java]: And why interface? Final class/enum would work - no one should "implement" `Headers`

**Review by grzegorz-robak — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java (hunk 1 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java]: The only benefit I see here would be with mocking. Downsides: another object to inject, longer syntax, state. I don't see that as an improvement in this case.

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java (hunk 2 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java]: So if the same header is used in both http and jms, there should be 2 classes with the same value?

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java (hunk 1 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java]: That's going into more academic discussion and I'd like to avoid that :) - so I'll just list my idea behind that comment:
- You don't have to inject anything - it's simple value class created with constructor
- syntax is longer (a little) but you won't have to manage *Utils classes (and they have the tendency to grow fast) 
- state is just the same string that you create with Util class so you have state already, just without object
- the improvement is just maintainability and readability (like with renaming variables)

To sum up - it's more like a preference so do as you wish

PS I hate Utils classes :)

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java (hunk 2 above)
Comment (grzegorz-robak) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/Headers.java]: Yeah, why not? They're completely different things, besides the name. What if one had to be renamed? Second approach: will worry about that when we have two the same headers

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java (hunk 1 above)
Comment (bartek-sarul) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/utils/OperationIdUtils.java]: Util classes are just because in java, even a static function must be inside a class.
I'll rename it to `OperationId` but leave inner function as static

**Review by grzegorz-robak — APPROVED** [review]
